rgw/admin/keys.go

```diff
@@ -0,0 +1,64 @@
+//go:build ceph_preview
+// +build ceph_preview
+
+package admin
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+)
+
+// KeySpec is the user credential management configuration
+// https://docs.ceph.com/en/latest/radosgw/adminops/#id37
+type KeySpec struct {
+	UID         string `url:"uid"`          // The user ID to receive the new key
+	SubUser     string `url:"subuser"`      // The subuser ID to receive the new key
+	KeyType     string `url:"key-type"`     // Key type to be generated, options are: swift, s3 (default)
+	AccessKey   string `url:"access-key"`   // Specify the access key
+	SecretKey   string `url:"secret-key"`   // Specify secret key
+	GenerateKey *bool  `url:"generate-key"` // Generate a new key pair and add to the existing keyring
+}
+
+// CreateKey will generate new keys or add specified to keyring
+// https://docs.ceph.com/en/latest/radosgw/adminops/#create-key
+// PREVIEW
+func (api *API) CreateKey(ctx context.Context, key KeySpec) (*[]UserKeySpec, error) {
+	if key.UID == "" && key.SubUser == "" {
```

Comment:
Should UID and SubUser be treated with different errors instead?

Reply:
I suppose it's just the same error - if neither UID neither nor SubUser provided it's an error, if any exists it's ok